scripts/sct_extract_metric.py

```diff
@@ -55,169 +57,203 @@ def get_parser():
---------------------------------------------------------------------------------------
-""" + label_references + """
---------------------------------------------------------------------------------------
-"""
-
-    parser.usage.addSection(str_section)
+        description += (f"\nTo compute average MTR in a region defined by a single label file (could be binary or 0-1 "
+                        f"weighted mask) between slices 1 and 4: {os.path.basename(__file__)} -i mtr.nii.gz -f "
+                        f"my_mask.nii.gz -z 1:4 -m wa\n"
+                        f"List of labels in {file_label}:\n"
+                        f"--------------------------------------------------------------------------------------\n"
+                        f"{label_references}\n"
+                        f"--------------------------------------------------------------------------------------\n")
+
+    parser = argparse.ArgumentParser(
+        description=description,
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Image file to extract metrics from. Example: FA.nii.gz"
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-f',
+        metavar=Metavar.folder,
+        default=os.path.join("label", "atlas"),
+        help=(f"Single label file, or folder that contains WM tract labels."
+              f"Example: {os.path.join(path_sct, 'data', 'atlas')}")
+    )
+    optional.add_argument(
+        '-l',
+        metavar=Metavar.str,
+        default='',
+        help=("Label IDs to extract the metric from. Default = all labels. Separate labels with ','. To select a group " 
+              "of consecutive labels use ':'. Example: 1:3 is equivalent to 1,2,3. Maximum Likelihood (or MAP) is "
+              "computed using all tracts, but only values of the selected tracts are reported.")
+    )
+    optional.add_argument(
+        '-method',
+        choices=['ml', 'map', 'wa', 'bin', 'max'],
+        default=param_default.method,
+        help=("R|Method to extract metrics.\n"
+              "  - ml: maximum likelihood (only use with well-defined regions and low noise)\n"
+              "    N.B. ONLY USE THIS METHOD WITH THE WHITE MATTER ATLAS! The sum of all tracts should be 1 in " 
+              "all voxels (the algorithm doesn't normalize the atlas).\n"
+              "  - map: maximum a posteriori. Mean priors are estimated by maximum likelihood within three clusters "
+              "         (white matter, gray matter and CSF). Tract and  noise variance are set with flag -p.\n"
```

Comment:
this line should be displayed with an indent (the indent added in the code does not appear in the terminal):
~~~
                          - map: maximum a posteriori. Mean priors are estimated by maximum likelihood within three clusters
                          (white matter, gray matter and CSF). Tract and  noise variance are set with flag -p.
                            N.B. ONLY USE THIS METHOD WITH THE WHITE MATTER ATLAS! The sum of all tracts should be 1 in all voxels (the
                            algorithm doesn't normalize the atlas).

~~~